[ButtonGroup] react and design guidelines (#293)
primer/brand: apps/docs/content/components/ButtonGroup/index.mdx

---
title: ButtonGroup
description: Use ButtonGroup to group related buttons together.
---

import ComponentLayout from '../../../src/layouts/component-layout'
export default ComponentLayout

![An image showing a ButtonGroup with a primary and a secondary button.](https://github.com/primer/brand/assets/912236/89ab4157-3c7a-4657-a7f0-1e481c943c41)

## Usage

Use ButtonGroup to group a set of actions that are related between them or to a single application. E,g, "Log in" and "Sign up" or "Try it out" and "Learn more".

> Use ButtonGroup to display a pair of primary and a secondary actions. The primary button is used for the main call to action while the secondary button is a related, less important action.

Reviewer: Is there a preferred order of primary and secondary?

Can I render more than 2 buttons in a button group? All of the examples only have 2 buttons shown. If only 2 buttons are supported, maybe we should consider renaming this component "ButtonPair".

> Use ButtonGroup to display a primary button with a secondary button, where the primary button should be the most important action and the secondary button should be a related action or a less important action.

Reviewer 2: ```suggestion
Use ButtonGroup to display a pair of primary and a secondary actions. The primary button is used for the main call to action while the secondary button is a related, less important action.
```

<DoDontContainer>
  <Do>
    <img src="https://github.com/primer/brand/assets/912236/f1b724f6-c10d-43b1-9e1f-bcdfc461d885" />
    <Caption>Use a primary button with a secondary button.</Caption>
  </Do>
  <Dont>
    <img src="https://github.com/primer/brand/assets/912236/d8880c67-8165-402b-ae16-07a5ec79efe6" />
    <Caption>Don’t place multiple primary buttons together.</Caption>
  </Dont>
</DoDontContainer>

<DoDontContainer>
  <Do>
    <img src="https://github.com/primer/brand/assets/912236/5f0c4d9f-73fe-4464-bdda-576521588fb7" />
    <Caption>

>       Do place primary buttons at the beginning of a button group.

Reviewer 2: ```suggestion
      Place the primary button at the beginning of a button group.
```

    </Caption>
  </Do>
  <Dont>
    <img src="https://github.com/primer/brand/assets/912236/807bfc0e-c0f5-4792-a31a-b1208b913a0e" />

>     <Caption>Don't place primary buttons at the end of a button group.</Caption>

Reviewer 2: ```suggestion
    <Caption>Don't place the primary button at the end of a button group.</Caption>
```
